test/RelayClient.test.ts

```diff
@@ -361,14 +372,160 @@ describe('RelayClient', function () {
     });
 
     it('should return true if its the owner of the smart wallet', async function () {
-      const verify = await relayClient.isSmartWalletOwner(smartWalletAddress, owner);
+      const verify = await relayClient.isSmartWalletOwner(
+        smartWalletAddress,
+        owner
+      );
+
       expect(verify).to.be.true;
     });
 
     it('shoudl return false if its not the owner of the smart wallet', async function () {
-      const verify = await relayClient.isSmartWalletOwner(smartWalletAddress, '0x02');
+      const verify = await relayClient.isSmartWalletOwner(
+        smartWalletAddress,
+        '0x02'
+      );
+
       expect(verify).to.be.false;
     });
-    
+  });
+
+  describe('estimateTokenTransferGas', function () {
+    let relayClient: RelayClient;
+
+    let ierc20TransferStub: SinonStub;
+
+    beforeEach(function () {
+      relayClient = new RelayClient();
+
+      ierc20TransferStub = sandbox.stub();
+      const estimateGas = {
+        transfer: ierc20TransferStub,
+      };
+
+      const ierc20Stub = {
+        estimateGas,
+      } as unknown as Sinon.SinonStubbedInstance<IERC20>;
+
+      sandbox.stub(IERC20__factory, 'connect').returns(ierc20Stub);
+    });
+
+    it('Should return 0 if token contract is zero address', async function () {
+      const relayRequest = config.util.toObject(
+        FAKE_RELAY_REQUEST
+      ) as RelayRequest;
+      relayRequest.request.tokenContract = constants.AddressZero;
+
+      expect(
+        (
+          await relayClient.estimateTokenTransferGas({}, relayRequest)
+        ).toString()
+      ).to.be.equal(BigNumber.from(0).toString());
+    });
+
+    it('Should return 0 if token amount is 0', async function () {
+      const relayRequest = config.util.toObject(
+        FAKE_RELAY_REQUEST
+      ) as RelayRequest;
+      relayRequest.request.tokenAmount = BigNumber.from(0);
+
+      expect(
+        (
+          await relayClient.estimateTokenTransferGas({}, relayRequest)
+        ).toString()
+      ).to.be.equal(BigNumber.from(0).toString());
+    });
+
+    it('Should fail if it is a deploy and the smartWallet is missing', async function () {
+      const requestConfig = config.util.toObject(
+        FAKE_REQUEST_CONFIG
+      ) as RequestConfig;
+      requestConfig.smartWalletAddress = constants.AddressZero;
```

Comment:
Im not see where are we validating or testing if the smartWalletAddress is missing, since we are assigning a AddressZero, I remove this and the test fails, meaning that the smartWalletAddress can be empty and does not work as expected

Reply:
Done.